extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -28,28 +87,366 @@ export class LaterpayVendor {
    * @param {!AccessService} accessService
    */
   constructor(accessService) {
-    /** @private @const */
+    /** @const @private {!AccessService} */
     this.accessService_ = accessService;
-    // TODO: implement
+
+    /** @const @private {!Window} */
+    this.win_ = this.accessService_.win;
+
+    /** @const @private {!Document} */
+    this.doc_ = this.win_.document;
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = viewportForDoc(this.win_.document);
+
+    /** @const @private {!LaterpayConfigDef} */
+    this.laterpayConfig_ = this.accessService_.getAdapterConfig();
+
+    /** @const @private {!PurchaseConfigDef} */
+    this.purchaseConfig_ = null;
```

Comment:
I turned off type compilation on your extension to avoid distracting you too much. So I only comment on the structural things. When we are ready, I'll turn it on and do few minor fixes in one PR.